src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Deployment/ElasticIndexRebuildDeploymentSource.cs

```diff
@@ -0,0 +1,39 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Search.Elasticsearch.Core.Services;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Deployment
+{
+    public class ElasticIndexRebuildDeploymentSource : IDeploymentSource
+    {
+        private readonly ElasticIndexSettingsService _elasticIndexSettingsService;
+
+        public ElasticIndexRebuildDeploymentSource(ElasticIndexSettingsService elasticIndexSettingsService)
+        {
+            _elasticIndexSettingsService = elasticIndexSettingsService;
+        }
+
+        public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+        {
+            var elasticIndexRebuildStep = step as ElasticIndexRebuildDeploymentStep;
+
+            if (elasticIndexRebuildStep == null)
+            {
+                return;
+            }
+
+            var indexSettings = await _elasticIndexSettingsService.GetSettingsAsync();
+
+            var data = new JArray();
+            var indicesToRebuild = elasticIndexRebuildStep.IncludeAll ? indexSettings.Select(x => x.IndexName).ToArray() : elasticIndexRebuildStep.Indices;
```

Comment:
Why include all the indices when `"includeAll"` is already set? The JSON will just include index names that won't be used. Same for the reset step.

Reply:
fixed